langchain4j-core/src/main/java/dev/langchain4j/store/memory/chat/FileChatMemoryStore.java

```diff
@@ -0,0 +1,133 @@
+     * @param fileExtension    File extension to store messages.
+     */
+    public FileChatMemoryStore(String storageDirectory, String fileExtension) {
+        Path directoryPath = Paths.get(storageDirectory);
+        if (Files.exists(directoryPath)) {
+            if (!Files.isDirectory(directoryPath)) {
+                throw illegalArgument("Storage directory path exists but is a file: %s", storageDirectory);
+            }
+        } else {
+            try {
+                Files.createDirectories(directoryPath);
+            } catch (IOException e) {
+                throw runtime("Something is wrong, Failed to create folder %s", storageDirectory);
+            }
+            log.debug("Created directory {} to store messages", storageDirectory);
+        }
+        this.storageDirectory = storageDirectory;
+        this.fileExtension = fileExtension.startsWith(".") ? fileExtension : "." + fileExtension;
+    }
+
+    public String getStorageDirectory() {
+        return storageDirectory;
+    }
+
+    public String getFileExtension() {
+        return fileExtension;
+    }
+
+    @Override
+    public List<ChatMessage> getMessages(Object memoryId) {
+        Path memoryFilePath = Paths.get(storageDirectory, memoryId + fileExtension);
+        if (Files.notExists(memoryFilePath)) {
+            return Collections.emptyList();
+        }
+        byte[] bytes;
+        try {
+            bytes = Files.readAllBytes(memoryFilePath);
+        } catch (IOException e) {
+            log.warn("Loading messages from file {} failed.", memoryFilePath);
+            return Collections.emptyList();
+        }
+        String fileContent = new String(bytes, StandardCharsets.UTF_8);
+        return ChatMessageDeserializer.messagesFromJson(fileContent);
+    }
+
+    @Override
+    public void updateMessages(Object memoryId, List<ChatMessage> messages) {
+        Path memoryFilePath = Paths.get(storageDirectory, memoryId + fileExtension);
+        // TODO：Maybe it would be better to store it item by item.
+        String messagesToJson = ChatMessageSerializer.messagesToJson(messages);
+        try {
+            Files.write(memoryFilePath, messagesToJson.getBytes(StandardCharsets.UTF_8));
+        } catch (IOException e) {
+            throw runtime("Update messages to file {} failed.", memoryFilePath);
+        }
+    }
+
+    @Override
+    public void deleteMessages(Object memoryId) {
+        Path memoryFilePath = Paths.get(storageDirectory, memoryId + fileExtension);
+        try {
+            Files.deleteIfExists(memoryFilePath);
+        } catch (IOException e) {
+            log.warn("Delete memory file {} failed", memoryFilePath);
+        }
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+
+        private String storageDirectory = System.getProperty("java.io.tmpdir");
+        private String fileExtension = ".json";
```

Comment:
Is there a good reason to make this confiigurable? Asking becayse the serializer and deserializer are not configurable so simply having a way to set the file extension can be deceptive

Reply:
> Is there a good reason to make this confiigurable? Asking becayse the serializer and deserializer are not configurable so simply having a way to set the file extension can be deceptive

Sorry, I'll modify it.